edit_course/views.py

```diff
@@ -324,21 +329,122 @@ def get_form_kwargs(self):
 
     def form_valid(self, form):
         from .operations.clone import clone
-        instance = clone(
+        new_instance = clone(
             instance=self.instance,
             url=form.cleaned_data['url'],
+            name=form.cleaned_data['instance_name'],
             clone_teachers=form.cleaned_data['teachers'],
             clone_assistants=form.cleaned_data['assistants'],
             clone_usertags=form.cleaned_data['usertags'],
-            clone_categories=form.cleaned_data['categories'],
-            clone_modules=form.cleaned_data['modules'],
-            clone_chapters=form.cleaned_data['chapters'],
-            clone_exercises=form.cleaned_data['exercises'],
             clone_menuitems=form.cleaned_data['menuitems'],
             siskey=form.cleaned_data.get('sis'),
         )
+
+        if not all([settings.GITMANAGER_URL, form.cleaned_data.get('git_origin'), form.cleaned_data.get('git_branch')]):
+            # Do not create a new entry in Git manager
+            messages.success(self.request, _('COURSE_INSTANCE_CLONED'))
+            return self.redirect(new_instance.get_url('course-details'))
+
+        key = "{}_{}{}".format(self.course.code, form.cleaned_data['key_year'], form.cleaned_data['key_month'])
+
+        data = {
+            "key": key,
+            "remote_id": new_instance.id,
+            "update_automatically": form.cleaned_data['update_automatically'],
+            "email_on_error": form.cleaned_data['email_on_error'],
+            "git_origin": form.cleaned_data['git_origin'],
+            "git_branch": form.cleaned_data['git_branch'],
+        }
+
+        permissions = Permissions()
+        permissions.instances.add(Permission.WRITE, id=new_instance.id)
+        gitmanager_url = urllib.parse.urljoin(settings.GITMANAGER_URL, f"api/gitmanager/{key}/")
+
+        try:
+            response = aplus_post(
+                gitmanager_url,
+                permissions=permissions,
+                data=data,
+                headers={"Application": "application/json, application/*"},
+            )
+            if response.status_code == 200:
+                new_instance.configure_url = urllib.parse.urljoin(settings.GITMANAGER_URL, f"{key}/aplus-json")
+                new_instance.save()
+                gitmanager_hook_url = urllib.parse.urljoin(settings.GITMANAGER_URL, f"gitmanager/{key}/hook")
+                response = aplus_post(
+                    gitmanager_hook_url,
+                    permissions=permissions,
+                    data={"key": key},
```

Comment:
Is the webhook secret not needed when the HTTP request includes the JWT token? (I don't know since I haven't checked.)

Reply:
Yes, that is correct.